### PR TITLE
NN-3021 Move global search from react

### PR DESCRIPTION
**Pull request by jsrobertson**
image

### CODE REVIEW

File: src/redux/actions/actions.test.js [hunk 1]
```diff
@@ -181,19 +181,6 @@ describe('actions', () => {
     expect(actions.setActivityData([{ stuff: 'stuff' }])).toEqual(expectedAction)
   })
 
-  it('should create an action to set establishment roll block data', () => {
```
Comment (jsrobertson) [inline src/redux/actions/actions.test.js:184]: Unrelated but unused so deleted while I was at it.

File: backend/controllers/globalSearch.js [hunk 2]
```diff
@@ -1,56 +1,131 @@
...
-      return {
-        ...offender,
-        dateOfBirth: formattedDateOfBirth,
-        uiId,
+
+    try {
+      if (searchText && !errors.length) {
+        log.info(`In globalSearch, searchText=${searchText}`)
+
+        const dateOfBirthFilter = dobIsValid ? dateOfBirth.format('YYYY-MM-DD') : undefined
```
Comment (JamieDraperUK) [inline backend/controllers/globalSearch.js:68]: Would the nullish operator work here? ` dobIsValid ?? dateOfBirth.format('YYYY-MM-DD')`

File: backend/controllers/globalSearch.js [hunk 3]
```diff
@@ -1,56 +1,131 @@
...
+        results: prisonerResults.map(prisoner => ({
+          ...prisoner,
+          showUpdateLicenceLink: isLicencesUser && (prisoner.currentlyInPrison === 'Y' || isLicencesVaryUser),
+          showProfileLink:
+            (userCanViewInactive && prisoner.currentlyInPrison === 'N') || prisoner.currentlyInPrison === 'Y',
+          updateLicenceLink: `${licencesUrl}hdc/taskList/${prisoner.latestBookingId}`,
+        })),
+      })
+    } catch (error) {
+      if (error) logError(req.originalUrl, error, 'Sorry, the service is unavailable')
```
Comment (JamieDraperUK) [inline backend/controllers/globalSearch.js:124]: Needs to be the page name :+1:

File: integration-tests/integration/globalsearch/globalSearch.spec.js [hunk 4]
```diff
@@ -100,59 +100,121 @@ context('Global search', () => {
...
     cy.task('stubGlobalSearch')
-    cy.visit('/global-search-results?searchText=quimby')
-    const globalSearchPage = GlobalSearchPage.verifyOnResultsPage()
-    globalSearchPage.rows().should(rows => {
-      expect(rows).to.have.lengthOf(3)
-      expect(rows[1]).to.have.text('Quimby, FredA1234AC15/10/1977Leeds HMPQuimby, Fred')
-      expect(rows[2]).to.have.text('Quimby, ArthurA1234AA15/09/1976Moorland HMPQuimby, Arthur')
+    cy.visit('/global-search/results?searchText=quimby')
+
+    cy.get('[data-test="global-search-results-table"]').then($table => {
```
Comment (JamieDraperUK) [inline integration-tests/integration/globalsearch/globalSearch.spec.js:105]: Seems a bit inconsistent not using the globalSearchPage given it's available.

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: backend/controllers/globalSearch.js (hunk 2 above)
Comment (jsrobertson) [reply backend/controllers/globalSearch.js:68]: Unfortunately not, with the way we then call the offenderSearchApi endpoint.

**Review by JamieDraperUK — APPROVED** [review]
LGTM

**Pull request merged**